System.IO.Hashing/Crc32.cs

```diff
@@ -0,0 +1,100 @@
+﻿//
+// Copyright (c) .NET Foundation and Contributors
+// Portions Copyright (c) Microsoft Corporation.  All rights reserved.
+// See LICENSE file in the project root for full license information.
+//
+
+using System.Runtime.CompilerServices;
+
+namespace System.IO.Hashing
+{
+    /// <summary>
+    ///   Provides an implementation of the CRC-32 algorithm, as used in
+    ///   ITU-T V.42 and IEEE 802.3.
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     If the platform or target doesn't support Hardware Crc32 computation it will offer a software implementation.
```

Comment:
Just to double check, it means that this will come with all platforms and every device, right?

Reply:
Just like any other assembly: depends on a build option.

Reply:
Can we make sure we have it present for all platforms even if it's a pure software implementation on the native side?